optimum/habana/transformers/generation/utils.py

```diff
@@ -383,16 +495,32 @@ def generate(
             )
 
         if generation_config.static_shapes:
-            # token_idx is the current index in the generation process, it is incremented each time a new token is generated
-            model_kwargs["token_idx"] = torch.tensor(inputs_tensor.shape[-1], device=inputs_tensor.device)
             # Pad inputs to have static shapes during generation, this gives better performance than dynamic shapes on HPUs
-            inputs_tensor = torch.nn.functional.pad(
-                inputs_tensor, (0, generation_config.max_new_tokens), value=generation_config.pad_token_id
-            )
-            if model_kwargs["attention_mask"] is not None:
-                model_kwargs["attention_mask"] = torch.nn.functional.pad(
-                    model_kwargs["attention_mask"], (0, generation_config.max_new_tokens), value=0
+            # In encoder_decoder models, Inputs are already padded
+            if not self.config.is_encoder_decoder:
+                # token_idx is the current index in the generation process, it is incremented each time a new token is generated
+                model_kwargs["token_idx"] = torch.tensor(inputs_tensor.shape[-1], device=inputs_tensor.device)
+                inputs_tensor = torch.nn.functional.pad(
+                    inputs_tensor, (0, generation_config.max_length), value=generation_config.pad_token_id
```

Comment:
```suggestion
                    inputs_tensor, (0, generation_config.max_new_tokens), value=generation_config.pad_token_id
```
If we pad with `max_length`, the size of the tensor will be `2*input_length + max_new_tokens`.

Reply:
Also, Shouldn't the behavior be controlled from command line (or) generation_config by setting ignore_eos value rather than depending on lazy mode value? 
As the behavior is now conflicting with CPU/GPU runs.

Reply:
Sure, you can add a condition to set it to False for encoder-decoder models. Could you provide some throughput numbers for both configurations (True and False).
It's already possible to set it within the generation config. This is just a default initialization if it has not been set.